tests.py

```diff
@@ -33,8 +37,8 @@ def run(extra_args):
 
     nose.main(addplugins=[x() for x in plugins],
               defaultTest=default_test_modules,
-              argv=sys.argv + extra_args)
-
+              argv=sys.argv + extra_args,
+              env=env)
```

Comment:
This should also be done in `lib/matplotlib/__init__.py:test`.  Probably move the definition of the environment variables to a private function in `__init__.py` and then get them from here and the `test` function.